src/permutation/permutation.rs

```diff
@@ -52,7 +52,7 @@ impl Permutation {
     /// added to the system
     fn valid_variables(&self, variables: &[Variable]) -> bool {
         let results: Vec<bool> = variables
-            .into_par_iter()
+            .iter()
```

Comment:
Would it be worth to consider gating the use of rayon under `std`, and choosing that over standard iterators if possible? I'm not sure of the performance impact of going single-threaded so I don't know how much time we will lose here.

Reply:
I already benchmarked that. The only thing worth to keep for rayon is the `parallel` feature used in `bls12_381`. The rest was even making it slower. As @ZER0 already suspected.

Reply:
And the `parallel` feature is used always when we enable `std`. But the rest of stuff is not worth. At least with the latest benchmarks I did a month ago it wasn;t/